### DISCUSSION TITLE
My Next JS app does not work on older iPads (12.2) and throws errors on ES6 code

### DISCUSSION THREAD

**Discussion by rogitong** [vercel/next.js#58677] [Help]
### Summary

I am working on latest nextjs 14.0.2
Standard configuration.
No fancy stuffff.

Looks like older ipads (12.2) do not support fully ES6 and throw errors which I dont want to show my user instead adopt the code or show "incompatible" page. For exampe:

**node_modules_next_dist_esm_build_templates_app-page_9417c2.js:554 SyntaxError: Unexpected token '?'**

This one comes from /packages/react-refresh-utils/internal/ReactRefreshModule.runtime.ts

**_2b8923._.js:57 SyntaxError: Unexpected token '.'**

This one comes from my line of code if (error?.response?.data) 

**node_modules_tailwind-merge_dist_bundle-mjs_mjs_b854ac._.js:57 SyntaxError: Unexpected token '.'**

this one comes from /tailwind-merge

**Question:**

How to get rid of this errors? All of my packages are latest. Addressing through browserlist does not change the thing even after deleting .next folder.

1. Detect ES5 incompatibility on client side. It does not help. Error is still thrown. 

2. Change configuration of nextjs? How and why? NextJS should support ios 12.

3. Update npm? I have latest v18. It does not make sense.

4. Other solution?

Seems like a very common error to solve, can you help? I dont want to have customers with "blank page" on production.

**Comment by icyJoseph** [comment]
I guess you can tweak the https://nextjs.org/docs/architecture/supported-browsers#browserslist

What iOS version are you trying to target?

**Reply by rogitong** [reply]
Thanks Joseph

I don't really understand the concept of "targeting". My goal is to have app available to as many as possible iPads and it looks like the problem is simple ES6 non compatible code. You recommend to add to the list minimum requirements like Safari? 

Or maybe there is other way around? How to check if Nextjs is incompatible and redirect to every page without an error to "non compliant browser" page?

**Reply by rogitong** [reply]
Looks like a bigger problem. Even if i redirect to 'incompatible' page using middleware.ts  i still get this errors. Those errors of course are not present in modern browsers. What is the best way to show user that the browser is not compatible. 

It loads and shows Page and then throw error. 
Safari 12.2

How to prevent the error? This elements are all over the code, including NextJS code. 

![Screenshot 2023-11-20 at 16 50 14](https://github.com/vercel/next.js/assets/123209082/6309c731-2072-467b-913b-93eb2b44fadc)

**Reply by briansmiley** [reply]
Hey @rogitong , did you end up finding a solution to this? Having exactly the same issue in my Next app on iPad and I'm very unfamiliar with how to go about troubleshooting.
